server/sonar-server/src/main/java/org/sonar/server/computation/step/CommentMeasuresStep.java

```diff
@@ -90,6 +109,90 @@ public BiSumCounter createNewCounter() {
     }
   }
 
+  private static class EmptyCounter implements Counter<EmptyCounter> {
+
+    @Override
+    public void aggregate(EmptyCounter counter) {
+      // nothing to do
+    }
+
+    @Override
+    public void aggregate(FileAggregateContext context) {
+      // nothing to do
+    }
+  }
+
+  private static class DocumentationFormula implements Formula<DocumentationCounter> {
+
+    @Override
+    public DocumentationCounter createNewCounter() {
+      return new DocumentationCounter();
+    }
+
+    @Override
+    public Optional<Measure> createMeasure(DocumentationCounter counter, CreateMeasureContext context) {
+      return getMeasure(context, counter.getPublicApiValue(), PUBLIC_API_KEY)
+        .or(getMeasure(context, counter.getPublicUndocumentedApiValue(), PUBLIC_UNDOCUMENTED_API_KEY))
+        .or(getDensityMeasure(counter, context));
+    }
+
+    private static Optional<Measure> getMeasure(CreateMeasureContext context, Optional<Integer> metricValue, String metricKey) {
+      if (context.getMetric().getKey().equals(metricKey) && metricValue.isPresent() && context.getComponent().getType().isHigherThan(Component.Type.FILE)) {
+        return Optional.of(Measure.newMeasureBuilder().create(metricValue.get()));
+      }
+      return Optional.absent();
+    }
+
+    private static Optional<Measure> getDensityMeasure(DocumentationCounter counter, CreateMeasureContext context) {
+      if (context.getMetric().getKey().equals(PUBLIC_DOCUMENTED_API_DENSITY_KEY) && counter.getPublicApiValue().isPresent() && counter.getPublicUndocumentedApiValue().isPresent()) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Split this 181 characters long line (which is greater than 180 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00103)